src/pages/investments.tsx

```diff
@@ -0,0 +1,108 @@
+import { useState } from "react";
+import Nav from "@/components/Nav";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { AlertCircle, Plus, Minus } from "lucide-react";
+import { Alert, AlertDescription, AlertTitle } from "@/components/ui/alert";
+import { api } from "@/utils/api";
+
+export default function InvestmentsTracker() {
+  const [input, setInput] = useState("");
+  const [showAlert, setShowAlert] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const { data: investmentsAmount, refetch: refetchInvestments } =
+    api.investments.get.useQuery();
+  const { mutate: updateInvestments } = api.investments.update.useMutation({
+    onSuccess: async () => {
+      console.log("Investments updated successfully");
+      await refetchInvestments();
+    },
+    onError: (error) => {
+      console.error("Detailed error updating investments:", error);
+      setErrorMessage(
+        `An error occurred while updating investments: ${error.message}`,
+      );
+      setShowAlert(true);
+    },
+  });
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+  };
+
+  const handleUpdateInvestments = (operation: "add" | "subtract") => {
+    const amount = parseFloat(input);
+    if (!isNaN(amount)) {
+      const newAmount =
+        operation === "add"
+          ? (investmentsAmount ?? 0) + amount
+          : Math.max((investmentsAmount ?? 0) - amount, 0);
+      console.log("Attempting to update investments with amount:", newAmount);
```

Comment:
Typically when pushing code, there's not much need for console logs. There are a few exceptions, but generally, it's best to avoid them in production code.